### ISSUE TITLE
Checking of `zip` over iterable of iterables fails

### ISSUE THREAD

**Issue by mdboom** [microsoft/pyright#5158]
**Describe the bug**

When using zip over an iterable or iterables, the typecheck fails.

**To Reproduce**

The following fails with 1.1.309.  It passed with 1.1.308.

```python
a = [(1, 2), (3, 4)]
for x, y in zip(a, a):
    pass
```

```
$ python -m pyright minimal.py
minimal.py:3:5 - error: "object*" is not iterable
    "__iter__" method not defined (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
```

**Expected behavior**

I would expect this to pass the type checker.

**VS Code extension or command-line**

Commandline, version 1.1.309.

**Comment by Julian** [comment]
(This is #5126 which I think is fixed on `main` and waiting for release)

**Comment by erictraut** [comment]
Yes, this is a duplicate of #5126. It will be fixed in 1.1.310 (which will be published later today).
